src/care_radiology/api/dicom.py

```
@@ -181,6 +179,85 @@ def upload(self, request):
+        permission_classes=[StaticAPIKeyAuthorization],
+    )
+    def upload_with_key(self, request):
+        patient = Patient.objects.get(external_id=request.data.get("patient_id"))
+        dcm_file = request.FILES.get("file")
+
+        if not dcm_file:
+            return Response({"error": "No file provided"}, status=400)
+
+        try:
+            body, content_type = encode_file_multipart_related(dcm_file)
+            upload_response = requests.post(
```

Ruff (0.15.17)

[error] Probable use of `requests` call without timeout

(S113)